webapp/components/form/Dropdown/Dropdown.js

```diff
@@ -1,143 +1,96 @@
+    <components.Option {...props} />
+  </div>
+)
 
-import ItemDialog from './ItemDialog'
-import { useLocalState, State } from './store'
+const IndicatorsContainerComponent = (props) => (
+  <div data-testid={TestId.dropdown.toggleBtn(props.selectProps.inputId)}>
+    <components.IndicatorsContainer {...props} />
+  </div>
+)
 
 const Dropdown = (props) => {
   const {
-    autocompleteDialogClassName,
     autocompleteMinChars,
     className,
+    clearable,
     disabled,
-    idInput,
+    id,
+    idInput: idInputProp,
     itemLabel,
     itemKey,
-    items,
+    items: itemsProp,
     onBeforeChange,
-    onChange,
+    onChange: onChangeProp,
     placeholder,
     readOnly,
     readOnlyInput,
     selection,
-    sourceElement,
+    testId,
     title,
     validation,
   } = props
 
-  const dropdownRef = useRef(null)
-  const inputRef = useRef(null)
-
-  const itemLabelFunction = useCallback(
-    (item) => (itemLabel.constructor === String ? A.prop(itemLabel, item) : itemLabel(item)),
-    [itemLabel]
-  )
-
-  const { state, Actions } = useLocalState({
-    autocompleteMinChars,
-    disabled,
-    itemKey,
-    itemLabelFunction,
-    items,
-    onBeforeChange,
-    onChange,
-    readOnly,
-    selection,
-    title,
-  })
-
-  // update itemLabelFunction and input value on itemLabelFunction change
-  useEffect(() => {
-    Actions.updateItemLabelFunction({ itemLabelFunction, selection })
-  }, [itemLabelFunction])
-
-  const showDialog = State.getShowDialog(state)
-  const itemsDialog = State.getItemsDialog(state)
-  const inputValue = State.getInputValue(state) || ''
-
-  const searchMinCharsReached = autocompleteMinChars <= 0 || inputValue.trim().length >= autocompleteMinChars
+  const { inputId, loading, menuIsOpen, onChange, onInputChange, openMenuOnClick, options, selectRef, value } =
+    useDropdown({
+      autocompleteMinChars,
+      idInputProp,
+      itemKey,
+      itemLabel,
+      itemsProp,
+      onBeforeChange,
+      onChangeProp,
+      selection,
+      title,
+    })
 
   return (
-    <div
-      ref={dropdownRef}
-      className={classNames('dropdown', className)}
-      onBlur={(event) => {
-        const { className: classNameTarget = '' } = event.relatedTarget || {}
-        if (classNameTarget !== ItemDialog.className) {
-          Actions.closeDialog({ selection })
-        }
-      }}
+    <ValidationTooltip
+      className="dropdown-wrapper dropdown-validation-tooltip"
+      id={id}
+      testId={testId}
+      validation={validation}
     >
-      <Input
-        id={idInput}
-        ref={inputRef}
+      <ReactSelect
+        className={classNames('dropdown', className)}
+        classNamePrefix="dropdown"
+        components={{ Option: OptionComponent, IndicatorsContainer: IndicatorsContainerComponent }}
+        inputId={inputId}
+        isClearable={clearable && !readOnly}
+        isDisabled={disabled}
+        isLoading={loading}
+        isSearchable={!readOnlyInput && !readOnly}
```

Comment:
do we have any way to put this false if we have an input and we don't like to have that like searchable?

Reply:
readOnlyInput=true?

Reply:
readOnlyInput and searchable are not comparable

readOnlyInput:  is always not searchable and not mutable
but
searchable:  is mutable and searchable 
but ! searchable:  could be mutable or not and not searchable 

as you have readOnly maybe readOnlyInput is not needed, true that makes me confused. why readOnly and readOnlyInput? what is the difference?

Reply:
readOnlyInput in this case means that you cannot write anything in the input field, and I think it's basically the same thing of using isSearchable (the opposite of it actually), while readOnly means that the dropdown is disabled, but it won't be rendered with a disabled style...

Reply:
ah another thing, readOnlyInput is always not searchable and could be mutable or not, like searchable.
Maybe it would be easier to change directly readOnlyInput into searchable and pass it to the react-select component, in the way we use it it won't make any difference

Reply:
readOnlyInput -> is not mutable and then not searchable, 
but !readOnlyInput could be searchable or not 

______________| w  | r | s | ( write, read, search )
readOnlyInput__|   0 1 0   | ( you can't type so !searchable)
!readOnlyInput_.|   1 1 x    | ( you can or can't have the possibility to search)
searchable_____|   1 1 1    |
!searchable____.|   x 1 0   | ( you can or can't have the possibility to edit the field)

for the searchable property:
readOnlyInput = !searchable  => 0 = 0 
but
!readOnlyInput =/= searchable => X ?= 1

To be comparable and to ensure that they are the opposite, It is necessary that the following be true:
A = !B => !A = B 
and this case is not totally true due to the previous point

Therefore maybe it is better to use isSearchable or searchable alone instead of mixing that with readability.

isSearchable={ isSearchable }